cmake/AJACommonDefines.cmake

```diff
@@ -21,12 +21,14 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
         -D_CRT_SECURE_NO_WARNINGS
         -D_SCL_SECURE_NO_WARNINGS)
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    add_definitions(
+    list(APPEND AJANTV2_COMPILE_DEFINITIONS
         -DAJALinux
         -DAJA_LINUX)
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    add_definitions(
+    list(APPEND AJANTV2_COMPILE_DEFINITIONS
         -DAJAMac
         -DAJA_MAC
         -D__STDC_CONSTANT_MACROS)
 endif()
+
+add_definitions(${AJANTV2_COMPILE_DEFINITIONS})
```

Comment:
[target_compile_definitions](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) is preferred in modern cmake if you want to attach those to specific targets (and they'll be exported alongside it)

Reply:
We use `target_compile_definitions` for the target-specific definitions in the repo. However I need to refactor these global defines at some point across all of our targets to apply them to targets with `target_compile_definitions`.